apis/nucleus/src/object/create-object.js

```diff
@@ -0,0 +1,58 @@
+import populateData from './populator';
+import { modelStore } from '../stores/model-store';
+
+/**
+ * @typedef {string | EngineAPI.INxDimension | EngineAPI.INxMeasure | LibraryField} Field
+ */
+
+/**
+ * @interface CreateConfig
+ * @description Rendering configuration for creating and rendering a new object
+ * @property {string} type
+ * @property {string=} version
+ * @property {(Field[])=} fields
+ * @property {EngineAPI.IGenericObjectProperties=} properties
+ */
+export default async function createObject(
+  { type, version, fields, properties, extendProperties /* , options, plugins, element */ },
+  halo,
+  generateOnly
+) {
+  let mergedProps = {};
+  // let error;
+  try {
+    const t = halo.types.get({ name: type, version });
+    mergedProps = await t.initialProperties(properties, extendProperties);
+    const sn = await t.supernova();
+    if (fields) {
+      populateData(
+        {
+          sn,
+          properties: mergedProps,
+          fields,
+        },
+        halo
+      );
+    }
+    if (properties && sn && sn.qae.properties.onChange) {
+      sn.qae.properties.onChange.call({}, mergedProps);
+    }
```

Comment:
What is this?

Reply:
Good question, it runs a onChange callback that you can put on you qae def. I think it is only used in a single place: https://github.com/qlik-oss/sn-mekko-chart/blob/master/src/index.js#L47
That thing updates coloring props when switching dims/meas.